go/vt/vtgate/planbuilder/gen4_planner.go

```diff
@@ -161,22 +180,25 @@ func checkUnsupportedConstructs(sel *sqlparser.Select) error {
 	return nil
 }
 
-func planHorizon(ctx planningContext, plan logicalPlan, sel *sqlparser.Select) (logicalPlan, error) {
-	hp := horizonPlanning{
-		sel: sel,
-	}
+func planHorizon(ctx planningContext, plan logicalPlan, in sqlparser.SelectStatement) (logicalPlan, error) {
+	sel, isSel := in.(*sqlparser.Select)
+	if isSel {
```

Comment:
We are already doing a `*sqlparser.Select` type check at line `114` before calling this method, I don't think we need to do it again here.

Reply:
we are also calling this method from `transformDerivedPlan`, and there we don't know the type, right?

Reply:
right, my bad!